### PR TITLE
fix:State name changes to report modals, view expense pages || Simplified report closure

### PR DESCRIPTION
**Pull request by arjunaj5**
https://app.clickup.com/t/85zrn14z8
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=gEJ1Iu3)

Loom recording of the changes: https://www.loom.com/share/33205ac80b2c48d19f1fb7dcf814d71e
Changes:

1. State name changes in add to report modal and add transactions to report modal
2. Expense state name changes across th view expense, mileage, and per diem pages.

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 1]
```diff
@@ -345,6 +345,8 @@ export class AddEditExpensePage implements OnInit {
 
   hardwareBackButtonAction: Subscription;
 
+  isNewReportsFlowEnabled = false;
```
Comment (yksolanki9) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:348]: Set this to `false` inside `ionViewWillEnter` as well

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 2]
```diff
@@ -217,6 +217,8 @@ export class AddEditMileagePage implements OnInit {
 
   hardwareBackButtonAction: Subscription;
 
+  isNewReportsFlowEnabled = false;
```
Comment (yksolanki9) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:220]: Same as above

**Review by yksolanki9 — APPROVED** [review]
Set `isNewReportsFlowEnabled` to false inside ionViewWillEnter as well. This is because if the page is stacked, member variables are not re-initialized, only `ionViewWillEnter` will run.

**Comment by arjunaj5** [comment]
Fixed the comments @yksolanki9 , please re-review

**Review by yksolanki9 — APPROVED** [review]

File: src/app/fyle/add-edit-expense/add-edit-expense.page.html [hunk 3]
```diff
@@ -952,6 +952,7 @@
                     [label]="etxn.tx.report_id ? 'Report' : 'Add to Report'"
                     [autoSubmissionReportName]="autoSubmissionReportName$|async"
                     [options]="reports"
+                    [isNewReportsFlowEnabled]="isNewReportsFlowEnabled"
```
Comment (gokulk16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.html:955]: simplifyReportsEnabled

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 1 above)
Comment (gokulk16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:348]: simplifyReportsEnabled

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 4]
```diff
@@ -2273,6 +2275,7 @@ export class AddEditExpensePage implements OnInit {
   }
 
   ionViewWillEnter() {
+    this.isNewReportsFlowEnabled = false;
```
Comment (gokulk16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2278]: simplifyReportsEnabled

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.html [hunk 5]
```diff
@@ -530,6 +530,7 @@
                             [options]="reports"
                             formControlName="report"
                             [autoSubmissionReportName]="autoSubmissionReportName$|async"
+                            [isNewReportsFlowEnabled]="isNewReportsFlowEnabled"
```
Comment (gokulk16) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.html:533]: simplifyReportsEnabled

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.html [hunk 6]
```diff
@@ -568,6 +568,7 @@
                                 [options]="reports"
                                 formControlName="report"
                                 [autoSubmissionReportName]="autoSubmissionReportName$|async"
+                                [isNewReportsFlowEnabled]="isNewReportsFlowEnabled"
```
Comment (gokulk16) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.html:571]: simplifyReportsEnabled

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts [hunk 7]
```diff
@@ -194,6 +194,8 @@ export class AddEditPerDiemPage implements OnInit {
 
   hardwareBackButtonAction: Subscription;
 
+  isNewReportsFlowEnabled = false;
```
Comment (gokulk16) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:197]: simplifyReportsEnabled

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts [hunk 8]
```diff
@@ -787,6 +790,10 @@ export class AddEditPerDiemPage implements OnInit {
       map((orgSettings) => orgSettings.per_diem.enable_individual_per_diem_rates)
     );
 
+    orgSettings$.subscribe((orgSettings) => {
```
Comment (gokulk16) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:793]: check if any other subscribe is present and we can add to that?

File: src/app/fyle/my-expenses/my-expenses.page.ts [hunk 9]
```diff
@@ -178,6 +178,8 @@ export class MyExpensesPage implements OnInit {
 
   hardwareBackButton: Subscription;
 
+  isNewReportsFlowEnabled = false;
```
Comment (gokulk16) [inline src/app/fyle/my-expenses/my-expenses.page.ts:181]: simplifyReportsEnabled

File: src/app/fyle/view-per-diem/view-per-diem.page.ts [hunk 10]
```diff
@@ -78,6 +78,8 @@ export class ViewPerDiemPage implements OnInit {
 
   projectFieldName: string;
 
+  isNewReportsFlowEnabled = false;
```
Comment (gokulk16) [inline src/app/fyle/view-per-diem/view-per-diem.page.ts:81]: simplifyReportsEnabled

File: src/app/shared/components/fy-add-to-report/fy-add-to-report-modal/fy-add-to-report-modal.component.ts [hunk 11]
```diff
@@ -46,6 +46,8 @@ export class FyAddToReportModalComponent implements OnInit, AfterViewInit {
 
   @Input() autoSubmissionReportName: string;
 
+  @Input() isNewReportsFlowEnabled = false;
```
Comment (gokulk16) [inline src/app/shared/components/fy-add-to-report/fy-add-to-report-modal/fy-add-to-report-modal.component.ts:49]: simplifyReportsEnabled

File: src/app/shared/components/fy-add-to-report/fy-add-to-report.component.ts [hunk 12]
```diff
@@ -46,6 +46,8 @@ export class FyAddToReportComponent implements OnInit, OnChanges, ControlValueAc
 
   @Input() autoSubmissionReportName: string;
 
+  @Input() isNewReportsFlowEnabled = false;
```
Comment (gokulk16) [inline src/app/shared/components/fy-add-to-report/fy-add-to-report.component.ts:49]: simplifyReportsEnabled

File: src/app/shared/components/fy-add-to-report/fy-add-to-report.component.ts [hunk 13]
```diff
@@ -110,6 +112,7 @@ export class FyAddToReportComponent implements OnInit, OnChanges, ControlValueAc
         subheader: this.subheader,
         enableSearch: this.enableSearch,
         autoSubmissionReportName: this.autoSubmissionReportName,
+        isNewReportsFlowEnabled: this.isNewReportsFlowEnabled,
```
Comment (gokulk16) [inline src/app/shared/components/fy-add-to-report/fy-add-to-report.component.ts:115]: simplifyReportsEnabled

File: src/app/shared/pipes/expense-state.pipe.ts [hunk 14]
```diff
@@ -4,19 +4,19 @@ import { Pipe, PipeTransform } from '@angular/core';
   name: 'expenseState',
 })
 export class ExpenseState implements PipeTransform {
-  transform(val) {
+  transform(val, isNewReportsFlowEnabled = false) {
```
Comment (gokulk16) [inline src/app/shared/pipes/expense-state.pipe.ts:7]: simplifyReportsEnabled

**Pull request merged**